crates/core/serde_utils.rs

```diff
@@ -30,6 +30,48 @@ pub mod u256 {
         let value = String::deserialize(d)?;
         U256::from_dec_str(&value).map_err(|e| D::Error::custom(e.to_string()))
     }
+
+    pub fn deser_hex_str<'de, D>(d: D) -> Result<U256, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value = String::deserialize(d)?;
+        U256::from_str_radix(value.trim_start_matches("0x"), 16)
+            .map_err(|_| D::Error::custom("Failed to deserialize u64 to u256 value"))
+    }
+
+    pub fn deser_hex_or_dec_str<'de, D>(d: D) -> Result<U256, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value = String::deserialize(d)?;
+        if value.starts_with("0x") {
+            U256::from_str_radix(value.trim_start_matches("0x"), 16)
+                .map_err(|_| D::Error::custom("Failed to deserialize u64 to u256 value"))
+        } else {
+            U256::from_dec_str(&value).map_err(|e| D::Error::custom(e.to_string()))
+        }
+    }
+}
+
+pub mod h160 {
+    use super::*;
+
+    use ethereum_types::H160;
+    use std::str::FromStr;
+
+    pub fn deser_hex_str<'de, D>(d: D) -> Result<H160, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value = String::deserialize(d)?;
+        if value.is_empty() {
+            Ok(H160::zero())
+        } else {
+            H160::from_str(value.trim_start_matches("0x"))
+                .map_err(|_| D::Error::custom("Failed to deserialize H160 value"))
+        }
+    }
```

Comment:
This implementation is misleading, while this is the behaviour we want for a `TxKind` field where an empty string is a valid value, we wouldn't want to use it to deserialize an address or H160 value